MatrixSDK/Background/MXBackgroundSyncService.swift

```diff
@@ -144,6 +143,11 @@ public enum MXBackgroundSyncServiceError: Error {
                         completion: @escaping (MXResponse<MXEvent>) -> Void) {
         NSLog("[MXBackgroundSyncService] fetchEvent: \(eventId). allowSync: \(allowSync)")
         
+        // Check local stores on every request so that we use up-to-data data from the MXSession store
+        if allowSync {
+            updateBackgroundServiceStoresIfNeeded()
```

Comment:
It may be a good idea to move this in `launchBackgroundSync`, as there are other cases triggering background sync.

Reply:
> Do you think we need to call this method for every call of `launchBackgroundSync`?

Yes, just in case for parallel session syncs.